### DISCUSSION TITLE
Pydantic models do not validate in the way the should when using `List[constr]`

### DISCUSSION THREAD

**Discussion by sammaphey** [fastapi/fastapi#7114] [Questions]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from typing import Any, List, Optional
from pydantic import BaseModel, constr, Field
from fastapi import FastAPI

path_regex = r'^(\/[^\/]+)+$'

class PatchOperation(BaseModel):
    """Definition for patch operation."""

    op: str
    path: constr(regex=path_regex)
    value: Optional[Any]
    from_: Optional[constr(regex=path_regex)] = Field(alias='from')

app = FastAPI()


@app.patch("/")
def read_root(
    operations: List[PatchOperation],
):
    print(operations)
    # I expect this print to show a List of PatchOperation objects as detailed [here](https://fastapi.tiangolo.com/tutorial/body/). 
    # Instead I get a List of dicts.
    return {"Hello": "World"}
```

### Description

* Open the swagger docs browser and call the patch endpoint `/`.
* It prints a dictionary from the server side output (rather than a PatchOperation object).
* It does not validate correctly if I pass in a path that does not match the regular expression. For example:
```json
[
    {
        "op": "replace",
        "path": "",
        "value": "Some new name"
    }
]
```
This input body should fail validation because my path_regex expects the string to start with a forward slash. When I test this out using Pydantic it works as advertised and does fail the validation.

### Environment

* OS: macOS:
* FastAPI Version: 0.61.1

* Python version: 3.7.7

### Additional context

It seem like the model is just not being utilized by FastAPI for some reason. I suspect that if the printed value of my input were actually a pydantic model then the validation would actually occur as expected

**Comment by sammaphey** [comment]
This is a lie. I am an idiot.

**Comment by tiangolo** [comment]
Thanks for closing the issue :+1:
